### PR TITLE
UIREC-35 Edit or receive piece

### PR DESCRIPTION
**Pull request by AndreiShumski2**
## Purpose
https://issues.folio.org/browse/UIREC-35
## Screenshots
image
image
## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.
- [x] I've added appropriate record to the CHANGELOG.md
- Does this PR meet or exceed the expected quality standards?
  - [ ] Code coverage on new code is 80% or greater
  - [x] Duplications on new code is 3% or less
  - [x] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [x] If any API-related changes - okapi interfaces and permissions are reviewed/changed correspondingly
  - [x] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/TitleDetails/ReceivingModal/ReceivingModal.js [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+        onClick={close}
+      >
+        <FormattedMessage id="ui-receiving.piece.actions.cancel" />
+      </Button>
+    </ModalFooter>
+  );
+
+  return (
+    <Modal
+      enforceFocus={false}
```
Comment (aliaksei-chumakou) [inline src/TitleDetails/ReceivingModal/ReceivingModal.js]: is there a reason to not enforce focus here?

File: src/TitleDetails/TitleDetailsContainer.js [hunk 2]
```diff
@@ -34,18 +37,44 @@ const TitleDetailsContainer = ({ location, history, mutator, match }) => {
...
 
-      mutator.pieces.GET({
+      return mutator.pieces.GET({
         params: {
           query: `poLineId==${lineId} sortby receiptDate`,
         },
       })
-        .then(setPieces)
-        .catch(() => setPieces([]));
+        .then((response) => {
```
Comment (aliaksei-chumakou) [inline src/TitleDetails/TitleDetailsContainer.js]: please give a readable name

File: src/TitleDetails/ReceivingModal/ReceivingModal.js (hunk 1 above)
Comment (AndreiShumski2) [reply src/TitleDetails/ReceivingModal/ReceivingModal.js]: removed it

File: src/TitleDetails/TitleDetailsContainer.js [hunk 3]
```diff
@@ -34,18 +37,44 @@ const TitleDetailsContainer = ({ location, history, mutator, match }) => {
...
+          });
+
+          return Promise.all([batchFetch(mutator.items, itemsIds), requestsPromise]);
+        })
+        .then(([pieceItems, itemRequests]) => {
+          setItems(pieceItems);
+          setRequests(itemRequests);
+        })
+        .catch(() => {
+          setPieces([]);
```
Comment (NikitaSedyx) [inline src/TitleDetails/TitleDetailsContainer.js]: why not to reset it before, you already call these setters without arguments

Comment (AndreiShumski2) [reply src/TitleDetails/TitleDetailsContainer.js]: where do you mean? it resets on the top of the method.

**Comment by AndreiShumski2** [comment]
Tests for receiving flow will be added in the next PR

**Pull request merged**